master/buildbot/steps/transfer.py

```diff
@@ -577,11 +579,15 @@ def __init__(self, o, workerdest=None,
         if workerdest is None:
             raise TypeError("__init__() takes at least 3 arguments")
 
+        self.super_class = StringDownload
         if 's' in buildstep_kwargs:
             del buildstep_kwargs['s']
-        s = json.dumps(o)
         StringDownload.__init__(
-            self, s=s, workerdest=workerdest, **buildstep_kwargs)
+            self, s=None, workerdest=workerdest, **buildstep_kwargs)
+
+    def start(self):
+        self.s = json.dumps(self.o)
+        return self.super_class.start(self)
```

Comment:
Because the other code in the file didn’t use `super` and I didn’t know why. I will change to `super` if there isn’t a reason for not doing it.